packages/mcdoc/src/runtime/attribute/builtin.ts

```diff
@@ -272,7 +285,46 @@ export function registerBuiltinAttributes(meta: core.MetaRegistry) {
 					const error = message
 						.replace(/^Invalid regular expression: /, '')
 						.replace(/^\/.+\/: /, '')
-					ctx.err.report(localize('invalid-regex-pattern', error), node, 2)
+					ctx.err.report(
+						localize('invalid-regex-pattern', error),
+						node,
+						core.ErrorSeverity.Warning,
+					)
+				}
+			}
+		},
+	})
+	registerAttribute(meta, 'match_regex', regexMatchValidator, {
+		checker: (config, typeDef, _) => {
+			if (typeDef.kind !== 'literal' || typeDef.value.kind !== 'string') {
+				return undefined
+			}
+			const pattern = config.pattern
+			const value = typeDef.value.value
+			return (node, ctx) => {
+				try {
+					const regex = RegExp(pattern)
+					if (!regex.test(value)) {
+						if (config.message) {
+							ctx.err.report(config.message, node, core.ErrorSeverity.Warning)
+						} else {
+							ctx.err.report(
+								localize('mismatching-regex-pattern', typeDef.value.value),
+								node,
+								core.ErrorSeverity.Warning,
+							)
+						}
+					}
+				} catch (e) {
+					const message = e instanceof Error ? e.message : `${e}`
+					const error = message
+						.replace(/^Invalid regular expression: /, '')
+						.replace(/^\/.+\/: /, '')
+					ctx.err.report(
+						localize('invalid-regex-pattern', error),
```

Comment:
I don't think this error should be reported. If the mcdoc type definitions contain a malformed regex pattern, the end-user shouldn't be bothered by that. Similar to some other attribute handling, this should just be ignored. At most it could be logged.

Reply:
`ctx.logger` messages shouldn't be translated, and also include some info that this is related to the `match_regex` mcdoc info. See the logs for `since` and `until` as reference.